Create script to generate icons (#1392)
steelthreads/agds-next: scripts/new-icon/README.md

# new-icon script

This script uses plop.js to generate the files and changes needed to add a new icon to AgDS.

> We prefer to use icons from [FeatherIcon](https://feathericons.com) to maintain a consistent style.

Reviewer: ```suggestion
We prefer to use icons from [Feather icons](https://feathericons.com) to maintain a consistent style.
```

Any SVG added to the icon set must meet this criteria:

> - Icon must be visually consistent with the rest of the suite

Reviewer: ```suggestion
- icon must be visually consistent with the rest of the suite
```

> - SVG Must be 24px x 24px

Reviewer: ```suggestion
- SVG dimensions must be 24px x 24px
```

- lines must be based on strokes and not outlined

> - SVG must be minified using [SVGOMG](https://jakearchibald.github.io/svgomg/)

Reviewer: ```suggestion
- when not using a feather icon, SVG must be cleaned up using [SVGOMG](https://jakearchibald.github.io/svgomg/)
```
